var/spack/repos/builtin/packages/py-keras/package.py

```diff
@@ -40,13 +46,80 @@ class PyKeras(PythonPackage):
     version('1.1.1', sha256='be1b67f62e5119f6f24a239a865dc47e6d9aa93b97b506ba34cab7353dbc23b6')
     version('1.1.0', sha256='36d83b027ba9d2c9da8e1eefc28f600ca93dc03423e033b633cbac9061af8a5d')
 
-    depends_on('python@3.6:', type=('build', 'run'), when='@2.4:')
-    depends_on('py-numpy@1.9.1:', type=('build', 'run'), when='@2.4:')
-    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4:')
-    depends_on('py-h5py', type=('build', 'run'), when='@2.4:')
-    depends_on('py-keras-applications', type='run', when='@2.2')
-    depends_on('py-keras-preprocessing', type='run', when='@2.2')
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.4')
+    depends_on('py-numpy@1.9.1:', type=('build', 'run'), when='@2.4')
+    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4')
+    depends_on('py-h5py', type=('build', 'run'), when='@2.4')
+    depends_on('py-keras-applications', type='run', when='@2.2:2.4')
+    depends_on('py-keras-preprocessing', type='run', when='@2.2:2.4')
     depends_on('py-setuptools', type='build')
     depends_on('py-theano', type=('build', 'run'), when='@:2.2')
-    depends_on('py-pyyaml', type=('build', 'run'))
+    depends_on('py-pyyaml', type=('build', 'run'), when='@:2.4')
     depends_on('py-six', type=('build', 'run'), when='@:2.2')
+    depends_on('py-tensorflow@2.5.0:2.5', type=('build', 'run'), when='@2.5.0:2.5')
+    depends_on('py-tensorflow@2.6.0:2.6', type=('build', 'run'), when='@2.6.0:2.6')
+    depends_on('py-tensorflow@2.7.0:2.7', type=('build', 'run'), when='@2.7.0:2.7')
+    depends_on('bazel', type='build', when='@2.5.0:')
+    depends_on('protobuf', type='build', when='@2.5.0:')
+
+    def url_for_version(self, version):
+        if version >= Version('2.6.0'):
+            url = 'https://github.com/keras-team/keras/archive/refs/tags/v{0}.tar.gz'
+            return url.format(version.dotted)
+        else:
+            return super(PyKeras, self).url_for_version(version)
+
+    @when('@2.5.0:')
+    def patch(self):
+        infile = join_path(self.package_dir, 'protobuf_build.patch')
+        with open(infile, 'r') as source_file:
+            text = source_file.read()
+        with open('keras/keras.bzl', mode='a') as f:
+            f.write(text)
+
+        filter_file('load("@com_google_protobuf//:protobuf.bzl", "py_proto_library")',
+                    'load("@org_keras//keras:keras.bzl", "py_proto_library")',
+                    'keras/protobuf/BUILD',
+                    string=True)
+
+    @when('@2.5.0:')
+    def build(self, spec, prefix):
```

Comment:
```suggestion
    def install(self, spec, prefix):
```
`PythonPackage` only has 1 phase now: `install`

Reply:
this is why i shouldn't attempt to multitask...